src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaseEquivalences.cs

```diff
@@ -131,34 +125,35 @@ static bool IsTurkishOrAzeri(ReadOnlySpan<char> cultureName)
         /// equivalent in a case comparison for 'A'.
         /// </remarks>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool TryFindCaseEquivalencesForChar(char c, out ReadOnlySpan<char> equivalences)
+        private static bool TryFindCaseEquivalencesForChar(char c, out ReadOnlySpan<char> equivalences)
         {
-            // Right shifting by 10 in order to divide by 1024 (CharactersPerRange)
-            Debug.Assert((c >> 10) < 0xFF);
-            byte index = (byte)(c >> 10);
-            ushort FirstLevelLookupValue = EquivalenceFirstLevelLookup[index];
+            // Dividing by 1024 whih represent the number of characters per range, in order to get the range index for c
+            Debug.Assert((c / 1024) < 0xFF);
+            byte index = (byte)((uint)c / 1024);
+            ushort firstLevelLookupValue = EquivalenceFirstLevelLookup[index];
 
             // If character belongs to a range that doesn't participate in casing, then just return false
-            if (FirstLevelLookupValue == 0xFFFF)
+            if (firstLevelLookupValue == 0xFFFF)
             {
                 equivalences = default;
                 return false;
             }
 
-            equivalences = PerformSecondLevelLookup();
+            equivalences = PerformSecondLevelLookup(c, firstLevelLookupValue);
             return equivalences != default;
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Why separate this out into a helper if it's only called from one place that aggressively inlines it?